src/site/content/en/blog/mediastreamtrack-insertable-media-processing/index.md

```diff
@@ -0,0 +1,249 @@
+---
+layout: post
+title: Insertable streams for MediaStreamTrack
+subhead: |
+  The contents of MediaStreamTrack objects exposed as a collection of streams so they can be manipulated
+  to introduce new components.
+authors:
+  - thomassteiner
+date: 2021-04-28
+description: |
+  Insertable streams for MediaStreamTrack expose the contents of a MediaStreamTrack as a collection
+  of streams (as defined by the WHATWG Streams API), which can be manipulated to introduce new
+  components.
+hero: image/8WbTDNrhLsU0El80frMBGE4eMCD3/Qu2wfQ3pxR8AeEfty88S.jpg
+alt: Cup of coffee and a laptop with a video conference running showing many participants.
+tags:
+  - blog # blog is a required tag for the article to show up in the blog.
+  - media
+  - capabilities
+---
+
+{% Aside %} Insertable streams for `MediaStreamTrack` is part of the
+[capabilities project](https://web.dev/fugu-status/) and is currently in development. This post will
+be updated as the implementation progresses. {% endAside %}
+
+## Background
+
+In the context of the
+[Media Capture and Streams API](https://developer.mozilla.org/en-US/docs/Web/API/Media_Streams_API)
+the [`MediaStreamTrack`](https://developer.mozilla.org/en-US/docs/Web/API/MediaStreamTrack)
+interface represents a single media track within a stream; typically, these are audio or video
+tracks, but other track types may exist as well.
+[`MediaStream`](https://developer.mozilla.org/en-US/docs/Web/API/MediaStream) objects consist of
+zero or more `MediaStreamTrack` objects, representing various audio or video tracks. Each
+`MediaStreamTrack` may have one or more channels. The channel represents the smallest unit of a
+media stream, such as an audio signal associated with a given speaker, like left or right in a
+stereo audio track.
+
+`MediaStream` objects have a single input and a single output. A `MediaStream` object generated by
+[`getUserMedia()`](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia) is
+called a local `MediaStream`, and has as its source input one of the user's cameras or microphones. A remote
+`MediaStream` may be representing to a media element, like `<video>` or `<audio>`, a stream
+originating over the network, and obtained via the WebRTC
+[`RTCPeerConnection`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection) API, or a
+stream created using the Web Audio API
+[`MediaStreamAudioSourceNode`](https://developer.mozilla.org/en-US/docs/Web/API/MediaStreamAudioSourceNode).
+
+## What is insertable streams for `MediaStreamTrack`?
+
+The core idea for insertable streams for `MediaStreamTrack` is to expose the content of a
+`MediaStreamTrack` as a collection of [streams](/streams/) (as defined by the WHATWG
+[Streams API](https://streams.spec.whatwg.org/)). These streams can be manipulated to introduce new
+components.
```

Comment:
```suggestion
components.

Granting developers access to the video (or audio) stream directly allows them to apply modifications
directly to the stream, rather than, for example, for the case of video manipulation,
force them to go through intermediates like `<canvas>` objects, as outlined in this
multi-step [article](https://html5doctor.com/video-canvas-magic/).
```